tools/dump_tool/HowToUseDumpTool.md

```diff
@@ -373,3 +492,11 @@ Like BaudRate, StopBits, Parity, Databits, HardwareFlowControl.
                 return -1;
         }
 ```
+## Abort parser vs dump parser tool
+
+1. The abort_parser prints all the valid addresses corresponding to the dump addresses in the stackdump retrieved from the dump logs using the addr2line command.
+
+2. The dump tool:
+a. Shows crash points corresponding to kernel/app crashes (using addr2line command).
+b. Shows symbols and addresses corresponding to the text address range in the stackdump for kernel/app stacks (using addr2line command).
+c. Shows the whole call stack corresponding to a crash in case of cortex R devices.
```

Comment:
Can we use this even with `cortex R`?
In `ramdumpParser.py`, there is a condition like below
```
		# Get ARM arch family
		if ('CONFIG_ARCH_FAMILY="armv8-m"' in data) or ('CONFIG_ARCH_FAMILY="armv7-m"' in data):
			#If architecture is cortex M, then run debugsymbolviewer script
                        #It displays the debug symbols corresponding to all the addresses in the kernel text address range
			os.system("python ../debug/debugsymbolviewer.py")
			return None
```
If we can use with cortext R, what is the difference for using this tool between cortex M and R?

Reply:
This is because we can't retrieve the whole call stack for cortex M.

00000000 <main>:
0:   e92d4800        push    {fp, lr}

Whereas we can retrieve it for cortex R >
 00000000 <main>:
 0:   b580            push    {r7, lr}